### ISSUE TITLE
Available Feature Exists? How to obtain the Captcha result in the handler (GraphQL)

### ISSUE THREAD

**Issue by bernarma** [chvarkov/google-recaptcha#43]
Not sure if this is something available and I haven't seen it in the documentation but is it possible to get the result of the validation in a query/mutation as a param to the function (similar to the "@CurrentUser" example on the NestJS website?

The use case I am trying to achieve is variable logic within the function based on the returned score. I.e. still perform the function but if the score is less than a specific value I would like to return a challenge to the user for an additional factor of authentication.

As copied from the Google Documentation
_reCAPTCHA v3 returns a score (1.0 is very likely a good interaction, 0.0 is very likely a bot). Based on the score, you can take variable action in the context of your site._

https://docs.nestjs.com/security/authentication#graphql

```
@Query(returns => User)
@UseGuards(GqlAuthGuard)
whoAmI(@CurrentUser() user: User) {
  return this.usersService.findById(user.id);
}
```

```
{
  "success": true|false,      // whether this request was a valid reCAPTCHA token for your site
  "score": number             // the score for this request (0.0 - 1.0)
  "action": string            // the action name for this request (important to verify)
  "challenge_ts": timestamp,  // timestamp of the challenge load (ISO format yyyy-MM-dd'T'HH:mm:ssZZ)
  "hostname": string,         // the hostname of the site where the reCAPTCHA was solved
  "error-codes": [...]        // optional
}
```

**Comment by chvarkov** [comment]
Hi @bernarma  thank you for this issue. It's interesting use case.
I will implement decorator that will work like you described.

**Comment by chvarkov** [comment]
Hi @bernarma 
This feature is published and available since 2.0.4 version.
Usage example
```
@Controller('feedback')
export class FeedbackController {
    @Recaptcha()
    @Post('send')
    async send(@RecaptchaResult() recaptchaResult: GoogleRecaptchaValidationResult): Promise<any> {
        console.log(recaptchaResult.action, recaptchaResult.score);
        // TODO: Your implementation.
    }
}
```

**Comment by bernarma** [comment]
I'll try it out later today and let you know how I go. Thanks for whipping this up... Much appreciated 👍

**Comment by bernarma** [comment]
I tested it just then... this is the error that is returned. I haven't looked at the error in too much detail (maybe later today) but is there anything that comes to mind?

If I remove the decorator it works again.

TypeError: this.loadModule is not a function
    at /Users/berniemac/Develop/projekt1/backend/node_modules/@nestlab/google-recaptcha/decorators/recaptcha-result.js:11:40
    at /Users/berniemac/Develop/projekt1/backend/node_modules/@nestjs/core/helpers/context-utils.js:36:28
    at resolveParamValue (/Users/berniemac/Develop/projekt1/backend/node_modules/@nestjs/core/helpers/external-context-creator.js:142:31)
    at Array.map (<anonymous>)
    at pipesFn (/Users/berniemac/Develop/projekt1/backend/node_modules/@nestjs/core/helpers/external-context-creator.js:145:45)
    at /Users/berniemac/Develop/projekt1/backend/node_modules/@nestjs/core/helpers/external-context-creator.js:68:23
    at InterceptorsConsumer.intercept (/Users/berniemac/Develop/projekt1/backend/node_modules/@nestjs/core/interceptors/interceptors-consumer.js:11:20)
    at target (/Users/berniemac/Develop/projekt1/backend/node_modules/@nestjs/core/helpers/external-context-creator.js:76:60)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async /Users/berniemac/Develop/projekt1/backend/node_modules/@nestjs/core/helpers/external-proxy.js:9:24

**Comment by Uter1007** [comment]
Get the same error when I am trying to use the decorator. Do I need to register it in the App.module as well ?

**Comment by Uter1007** [comment]
In my case I am using it in a Graphql Resolver but the same stacktrace will be printed out...

**Comment by bernarma** [comment]
> Get the same error when I am trying to use the decorator. Do I need to register it in the App.module as well ?

Using CurrentUser as the basis of my response, I have not needed to register this in the module.

**Comment by Uter1007** [comment]
@chvarkov could you give it a look ? Would love to log out the recaptcha score ;)

**Comment by chvarkov** [comment]
Hi @bernarma @Uter1007 
Thanks for feedback. It was my mistake, sorry.
Fix was published and available since 2.0.5 version.

**Comment by chvarkov** [comment]
@Uter1007 About importing module.
Yes, You need to import `GoogleRecaptchaModule` to `SharedModule` or `AppModule`. It's need for declaring module providers like Guard and internal services.
For GraphQL application you need to add option `applicationType: ApplicationType.GraphQL`.

In future will do application type detection dynamically.
